Tasks/Common/ServiceFabricHelpers/Get-ServiceFabricEncryptedText.ps1

```diff
@@ -12,15 +12,30 @@ function Get-ServiceFabricEncryptedText
 
     $defaultCertStoreName = "My"
     $defaultCertStoreLocation = "CurrentUser"
-    $serverCertThumbprint = $ClusterConnectionParameters["ServerCertThumbprint"]
+    $serverCertThumbprints = $ClusterConnectionParameters["ServerCertThumbprint"]
+
+    if ($serverCertThumbprints -is [array])
+    {
+        foreach($serverCertThumbprint in $serverCertThumbprints)
+        {
+            $cert = Get-Item "Cert:\$defaultCertStoreLocation\$defaultCertStoreName\$serverCertThumbprint" -ErrorAction SilentlyContinue
+            if (-not $cert)
```

Comment:
Why should it be -not $cert. Shouldn't it be the other way round. If a cert is found then only break from the loop.

Reply:
Good catch, added unit test to verify.